File: Csharp/CodeWars/src/Resolutions/ConsonantValue.cs
namespace CodeWars.Resolutions;

public class ConsonantValue
{
    public static int Solve(string s)
    {
        int currentSum = 0;
        int maxSum = 0;

        foreach (char ch in s.ToLower())
        {
            if (IsVowel(ch))
            {
                if (currentSum > 0)
                {
                    maxSum = Math.Max(maxSum, currentSum);
                    currentSum = 0;
                }
            }
            else if (char.IsLetter(ch))
                currentSum += AlphabetPosition(ch);
        }

        if (currentSum > 0)
            maxSum = Math.Max(maxSum, currentSum);

        return maxSum;
    }
    private static bool IsVowel(char c) =>
        "aeiou".Contains(c);

    private static int AlphabetPosition(char c) =>
        c - 'a' + 1;
}

File: TypeScript/Codewars/src/resolutions/Consonant-value.ts
export function solve(s: string): number {
	let currentSum = 0;
	const sums: number[] = [];
	const vowelsRegex = /[aeiou]/;

	for (const char of s.toLowerCase()) {
		if (vowelsRegex.test(char)) {
			if (currentSum > 0) sums.push(currentSum);
			currentSum = 0;
		} else {
			currentSum += calculateAlphabetPosition(char);
		}
	}

	if (currentSum > 0) sums.push(currentSum);

	return sums.length > 0 ? Math.max(...sums) : 0;
}

function calculateAlphabetPosition(char: string): number {
	const asciiValueForA = 97;
	return char.charCodeAt(0) - asciiValueForA + 1;
}
